front/app/modules/commercial/matomo/index.test.ts

```diff
@@ -0,0 +1,76 @@
+import config from '.';
+import { mockRoutes } from './matchPath.test';
+import eventEmitter from 'utils/eventEmitter';
+import { trackPage } from 'utils/analytics';
+
+jest.mock('services/appConfiguration');
+jest.mock('services/auth');
+jest.mock('services/locale');
+jest.mock('modules', () => ({ streamsToReset: [] }));
+jest.mock('routes', () => ({
+  __esModule: true,
+  default: jest.fn(() => [mockRoutes]),
+}));
+
+global.window = Object.create(window);
+Object.defineProperty(window, 'location', {
+  value: { pathname: '/en' },
+});
+
+describe('matomo', () => {
+  beforeEach(() => {
+    window._paq = [];
+  });
+
+  it('does not make any calls if no consent given', () => {
+    (config as any).beforeMountApplication();
+    eventEmitter.emit<any>('destinationConsentChanged', { matomo: false });
+
+    expect(window._paq).toEqual([]);
+  });
+
+  it('makes call for initial page after consent is given', () => {
+    (config as any).beforeMountApplication();
+    eventEmitter.emit<any>('destinationConsentChanged', { matomo: true });
+
+    const expectedCalls = [
+      ['enableLinkTracking'],
+      ['enableHeartBeatTimer'],
+      ['setDomains', 'wonderville.com/*'],
+      ['setCookieDomain', 'wonderville.com/*'],
+      ['setUserId', '522ae8cc-a5ed-4d31-9aa0-470904934ec6'],
+      ['setTrackerUrl', '//matomo.hq.citizenlab.co/matomo.php'],
+      ['setSiteId', '14'],
+      ['addTracker', '//matomo.hq.citizenlab.co/matomo.php', '13'],
+      ['setCustomDimension', 1, 'wonderville'],
+      ['setCustomDimension', 2, 'c4b400e1-1786-5be2-af55-40730c6a843d'],
+      ['setCustomDimension', 3, 'en'],
+      ['setCustomUrl', '/en'],
+      ['trackPageView', '/:locale'],
+      ['enableLinkTracking'],
+      ['enableHeartBeatTimer'],
+      ['setDomains', 'wonderville.com/*'],
+      ['setCookieDomain', 'wonderville.com/*'],
+      ['setUserId', '522ae8cc-a5ed-4d31-9aa0-470904934ec6'],
+      ['setTrackerUrl', '//matomo.hq.citizenlab.co/matomo.php'],
+      ['setSiteId', '14'],
+      ['addTracker', '//matomo.hq.citizenlab.co/matomo.php', '13'],
+      ['setCustomDimension', 1, 'wonderville'],
+      ['setCustomDimension', 2, 'c4b400e1-1786-5be2-af55-40730c6a843d'],
+      ['setCustomDimension', 3, 'en'],
+      ['setCustomUrl', '/en'],
+      ['trackPageView', '/:locale'],
+    ];
+
+    expect(window._paq).toEqual(expectedCalls);
+  });
+
+  it('tracks page change', () => {
+    (config as any).beforeMountApplication();
+    eventEmitter.emit<any>('destinationConsentChanged', { matomo: true });
+
+    trackPage('/en/sign-in');
+
+    console.log(window._paq)
```

Comment:
If you read the output of this `console.log` you'll see that the page change here is registered three times. Since that seems broken I didn't write an assertion for that yet

Reply:
Yes, I've just been manually testing this change. There is massive duplication happening across all the tracking, such that my local Matomo was telling me I'd logged 342 actions this morning when I'd merely clicked a few pages and logged in and out - https://docs.google.com/spreadsheets/d/1tYz5uCecvblG9CrBz0IfNj2LAlvOKdwnzVQ5kF5Fv-s/edit?usp=sharing